File: candy/candy.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        vector<int>res(ratings.size(),1);
        int ans=0,i;
        
        for ( i=1;i<ratings.size();i++){
            if (ratings[i] > ratings[i-1] and res[i] <= res[i-1]){
                res[i] = res[i-1]+1;
            }
        }
        
        //right check 
        for (i =ratings.size()-2;i>=0;i--){
            if(ratings[i] > ratings[i+1] and res[i]<=res[i+1]){
                res[i]= res[i+1]+1;
            }
        }
        
        
        for (i=0;i<res.size();i++){
            ans+=res[i];
            cout<<res[i]<<" ";
        }
        return ans;
    }
};
/*

    [1,2,3,4,4,3,2,1]

left 1 2 3 4 1 1 1 1
right 1  2  3  4  4 3  2  1







*/

File: candy/candy.java
class Solution {
    public int candy(int[] ratings) {
        int candies[]=new int[ratings.length];
        int candiesReq=0;
        Arrays.fill(candies,1);
        for (int loop=1;loop<ratings.length;loop++){
            if (ratings[loop]>ratings[loop-1] && candies[loop]<=candies[loop-1]){
                candies[loop]=candies[loop-1]+1;
            }
        }
        for (int loop=ratings.length-2;loop>=0;loop--){
            if (ratings[loop]>ratings[loop+1] && candies[loop]<=candies[loop+1]){
                candies[loop]=candies[loop+1]+1;
            }
        }
        for (int loop=0;loop<candies.length;loop++){
            candiesReq+=candies[loop];
        }
        return candiesReq;
    }
}